zerver/decorator.py

```diff
@@ -693,11 +694,11 @@ def _wrapped_func_arguments(
                 if not webhook_client_name:
                     raise err
                 if isinstance(err, JsonableError) and not isinstance(
-                    err, UnsupportedWebhookEventType
+                    err, (UnsupportedWebhookEventType, AnomalousWebhookPayload)
                 ):  # nocoverage
                     raise err
 
-                if isinstance(err, UnsupportedWebhookEventType):
+                if isinstance(err, (UnsupportedWebhookEventType, AnomalousWebhookPayload)):
```

Comment:
I don't think these changes are correct.  The intent is that `UnsupportedWebhookEventType` should not be returned to users as a 400 error, because the request from the third party is a valid event -- we just don't do anything for that event. 

Whereas for `AnomalousWebhookPayload`, it's a 400 error -- we think that you're sending requests from a thing that is not a Jira server to Zulip.

Reply:
I'm not sure how to do the plumbing to have `webhook_name` get set if we remove this; might require some fiddling.

Reply:
From a logging perspective, I think we want these to go to `webhooks_unexpected_events.log`, whereas the `UnsupportedWebhookEventType` should go to `webhooks_unsupported_events.log`.

```
webhooks_unexpected_events.log
webhooks_unsupported_events.log
```

Reply:
@timabbott As for your first comment, isn't `UnsupportedWebhookEventType` already being returned as a 400 error since the exception subclasses `JsonableError` with a default status code of 400? Note that `UnsupportedWebhookEventType`, when caught, is re-raised at the end after logging. Are you saying that this PR should make it such that `UnsupportedWebhookEventType` should not be re-raised after logging?

Moreover, I would say that both `AnomalousWebhookPayload` and `UnsupportedWebhookEventType` represent a 400 error in the sense that both are "bad requests" that we can't handle, either because we choose not to, or because the payload seems incorrectly formatted or from an incorrect service. Conceptually, even `UnsupportedWebhookEventType` should be presented as an error that the user can report so that if we get too many of these for one event type, we can add support for said event.

As for your comment about the log file name, I named it `webhooks_anomalous_payloads.log`, since the scope of the term `anomalous payloads` is wider than that for the term `unexpected events`, which could be misinterpreted as just being about service-specific webhook events, and not just about anomalous payloads in general. Thanks!

Reply:
I like `webhooks_anomalous_payloads.log` as a name, both because it matches the exception class, and also because it's harder than `unexpected` to confuse for `unsupported` at a glance. :)

Reply:
On the HTTP status code front, we are certainly returning 400's currently for `UnsupportedWebhookEventType` exceptions.  I think there's a potential argument to make those 501's, but [the specs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501) imply that this is generally treated as the _HTTP method_ being unimplemented, not the resource.

Reply:
501s don't seem appropriate here either, I agree. Honestly, I think 400s are fine for both of these, these are both classes of errors that should fall under the same larger category of stuff that can go wrong with our webhooks. Even if we choose to not support an event, that can still be an error in the sense that maybe we should if there are lots of these exceptions for the same type of payload/event.